modin/config/pubsub.py

```diff
@@ -224,21 +228,21 @@ def _get_default(cls):
         return cls.default
 
     @classmethod
-    def get_value_source(cls):
+    def get_value_source(cls) -> Optional[ValueSource]:
```

Comment:
I wonder why it's optional here, it shouldn't be returning `None`

Reply:
It's possible that if we try to call `get` and `cls._value` is not `_UNSET`, we can still have `cls._value_source` stay as `None` right? That's why `mypy` complained about it earlier.

Reply:
This feels like a bug in logic somewhere... if `cls._value` is set to something, its value source should be defined.

I think `mypy` complains here because `._value` is not a _private_ attribute, so in theory someone can set it from outside.
Maybe we should `assert cls._value_source is not None` before returning, so that `get_value_source()` returns non-optional?

Reply:
Yeah I think that makes sense. While we can have such a case, it's better that we catch and report the error via an assert instead of allowing that kind of behavior.